servers/rendering/rendering_device.cpp

```diff
@@ -168,7 +159,6 @@ void RenderingDevice::_free_dependencies(RID p_id) {
 		for (const RID &F : E->value) {
 			HashMap<RID, HashSet<RID>>::Iterator G = dependency_map.find(F);
 			ERR_CONTINUE(!G);
-			ERR_CONTINUE(!G->value.has(p_id));
```

Comment:
I agree with the other changes, but I'm wondering about this one. I'm not sure, but it looks as if the id being in the hash set was a precondition that the original author(s) considered important enough for integrity. In that case, this `ERR_CONTINUE` is akin to a runtime assertion but without the crashy consequences. If I'm right here, I think this check has to stay.

Reply:
I agree, this is a useful check to ensure the data is valid, it would be an indication of something being wrong if you're trying to erase the wrong data

Reply:
If we want to keep the check but potentially optimize we can instead do:
```cpp
bool erased = G->value.erase(p_id);
if (unlikely(!erased)) {
    ERR_PRINT(...);
}
```

Reply:
Yes, indeed, I did not notice this.